classes/services/UserService.inc.php

```diff
@@ -456,4 +467,136 @@ public function canCurrentUserGossip($userId) {
 
 		return true;
 	}
+
+	/**
+	 * Merge two user accounts, including attributed articles and all associated
+	 * information.
+	 *
+	 * @param $user User The user to merge from. This user will no longer exist
+	 *  when this operation is done.
+	 * @param $mergeIntoUser User The user to merge into. This user will contain
+	 *  all data from $user when this operation is done.
+	 */
+	function mergeUsers($user, $mergeIntoUser) {
+		$userId = $user->getId();
+		$mergeIntoUserId = $mergeIntoUser->getId();
+
+		HookRegistry::call('User::mergeUsers', array(&$userId, &$mergeIntoUserId));
+
+		$noteDao = DAORegistry::getDAO('NoteDAO');
+		$notes = $noteDao->getByUserId($userId);
+		while ($note = $notes->next()) {
+			$note->setUserId($mergeIntoUserId);
+			$noteDao->updateObject($note);
+		}
+
+		$editDecisionDao = DAORegistry::getDAO('EditDecisionDAO');
+		$editDecisionDao->transferEditorDecisions($userId, $mergeIntoUserId);
+
+		$reviewAssignmentDao = DAORegistry::getDAO('ReviewAssignmentDAO');
+		foreach ($reviewAssignmentDao->getByUserId($userId) as $reviewAssignment) {
+			$reviewAssignment->setReviewerId($mergeIntoUserId);
+			$reviewAssignmentDao->updateObject($reviewAssignment);
+		}
+
+		$articleEmailLogDao = DAORegistry::getDAO('SubmissionEmailLogDAO');
+		$articleEmailLogDao->changeUser($userId, $mergeIntoUserId);
+		$articleEventLogDao = DAORegistry::getDAO('SubmissionEventLogDAO');
+		$articleEventLogDao->changeUser($userId, $mergeIntoUserId);
+
+		$submissionCommentDao = DAORegistry::getDAO('SubmissionCommentDAO');
+		$submissionComments = $submissionCommentDao->getByUserId($userId);
+
+		while ($submissionComment = $submissionComments->next()) {
+			$submissionComment->setAuthorId($mergeIntoUserId);
+			$submissionCommentDao->updateObject($submissionComment);
+		}
+
+		$accessKeyDao = DAORegistry::getDAO('AccessKeyDAO');
+		$accessKeyDao->transferAccessKeys($userId, $mergeIntoUserId);
+
+		// Transfer old user's individual subscriptions for each journal if new user
```

Comment:
for OMP changes: consider this journal word :-)

Reply:
Oh, now that you mention it, this whole method is going to need to be split into app-specific methods.